File: backEnd/platform-automation/src/main/java/com/mokatest/platform/demos/api/domain/requestModel/AssertParameter.java
package com.mokatest.platform.demos.api.domain.requestModel;

import com.mokatest.platform.demos.api.domain.apiEnum.RuleSource;
import com.mokatest.platform.demos.enums.AssertRelationship;
import com.mokatest.platform.demos.api.domain.apiEnum.ApiAssertType;
import lombok.Data;

/**
 * @Description: 断言参数
 * @Author: JingLong
 * @DateTime: 2026/4/3 14:27
 */
@Data
public class AssertParameter {
    // 断言体
    private ApiAssertType apiAssertType;
    // 字段
    private String field;
    // 断言关系
    private AssertRelationship assertRelationship;
    // 断言值
    private String assertValue;
    // 是否禁用
    private boolean disabled = false;

    /**
     * 断言规则来源（运行时填充，不持久化到数据库）
     * 用于标识该断言规则来自 GLOBAL/ENVIRONMENT/SCENE/API
     */
    private RuleSource source;
}

File: frontEnd/src/types/domain/api/requestModel/AssertParameter.ts
import {ApiAssertType} from "@/types/domain/api/apiEnum/ApiAssertType";
import {AssertRelationship} from "@/types/enum/condation/AssertRelationship";

export class AssertParameter {
    // 断言体
    apiAssertType?: ApiAssertType;
    // 字段
    field?: string;
    // 断言关系
    assertRelationship?: AssertRelationship;
    // 断言值
    assertValue?: string;

    // 是否禁用
    disabled?: boolean;

    /**
     * 断言规则来源（运行时填充，不持久化到数据库）
     * GLOBAL | ENVIRONMENT | SCENE | API
     */
    source?: string;

    constructor() {
        this.apiAssertType = ApiAssertType.BODY,
            this.field = '',
            this.assertRelationship = AssertRelationship.EQUALS,
            this.assertValue = '',
            this.disabled = false
    }
}
